### PR TITLE
Use rollup to bundle the SDK

### LINKED ISSUE #32: Getting started doesn't work straightforward in a pure JS app
**Description:**
I have been trying to start a JS app by referring to the getting started section.
image

Above highlighted two lines didn't work in a pure JS app.
Instead, I had to use the following line.

```
var auth = AsgardioAuth.IdentityClient.getInstance();
```

Can we validate this and update the README.

### LINKED ISSUE #51: Create an enum for response mode types
Right now, the available options for response modes are defined by a type. However, they should be defined by an enum so that consumers of the SDK can also make use of it while configuring the SDK.

### LINKED ISSUE #53: Use rollup to bundle the JavaScript authentication SDK
**Is your suggestion related to an experience ? Please describe.**
As of now, we are using webpack to bundle the SDK. Since webpack doesn't natively support esm modules, we need to use rollup to bundle the SDK.

### LINKED ISSUE #62: Remove allow-access-control-origin header attribute from requests
**Describe the issue:**
$subject

### PR DESCRIPTION
**Pull request by thivi**
## The changes introduced by this PR
- Replacing webpack with rollup to bundle the SDK
- Creating three bundles for ESM, UMD, and browsers
- Creating a separate set of polyfilled bundles
- Adding sourcemaps to the bundles
- Ejecting the React sample app bootstrapped using Create React App. This was to avoid issues caused by eslint version mismatch
- Removing the `Access-Control-Allow-Origin` attribute from the header of requests
- Improving the Getting Started section of the readme
- Renaming the global variable from `AsgardioAuth` to `AsgardioOIDC`
- Updating the readme file with the information about browser compatibility
- Replacing `jsrsasign` library with `jose`
- Reducing the bundle file size of the default bundle to 145KB.

### CODE REVIEW

File: packages/oidc-js/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            return pkg.main;
+        case ESM_BUNDLE:
+            return pkg.module;
+        case BROWSER_BUNDLE:
+            return "dist/asgardio-oidc.production.min.js";
+        default:
+            return pkg.main;
+    }
+};
+/**
```
Comment (DimalChandrasiri) [inline packages/oidc-js/rollup.config.js:113]:
```suggestion

/**
```

File: samples/using-oidc-js-sdk/react-js-app/config/env.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    'process.env': Object.keys(raw).reduce((env, key) => {
+      env[key] = JSON.stringify(raw[key]);
+      return env;
+    }, {}),
+  };
+
+  return { raw, stringified };
+}
+
+module.exports = getClientEnvironment;
```
Comment (DimalChandrasiri) [inline samples/using-oidc-js-sdk/react-js-app/config/env.js:101]: Please replace `'` with `"`. Check other instances and files as well.

File: samples/using-oidc-js-sdk/react-js-app/config/env.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+'use strict';
```
Comment (DimalChandrasiri) [inline samples/using-oidc-js-sdk/react-js-app/config/env.js:1]: Shouldn't the sample files need the license headers as well?

**Review by DimalChandrasiri — APPROVED** [review]

File: samples/using-oidc-js-sdk/react-js-app/config/env.js (hunk 2 above)
Comment (thivi) [reply samples/using-oidc-js-sdk/react-js-app/config/env.js:115]: Fixed by 5c8fc5d

File: samples/using-oidc-js-sdk/react-js-app/config/env.js (hunk 3 above)
Comment (thivi) [reply samples/using-oidc-js-sdk/react-js-app/config/env.js]: Fixed by 5c8fc5d

**Pull request merged**